presto-orc/src/test/java/com/facebook/presto/orc/TestTupleDomainFilter.java

```diff
@@ -363,12 +370,39 @@ public void testMultiRange()
 
         filter = MultiRange.of(ImmutableList.of(
                 FloatRange.of(Float.MIN_VALUE, true, true, 1.2f, false, true, false),
-                FloatRange.of(1.2f, false, true, Float.MAX_VALUE, true, true, false)), false);
+                FloatRange.of(1.2f, false, true, Float.MAX_VALUE, true, true, false)), false, false);
 
         assertTrue(filter.testFloat(1.1f));
         assertTrue(filter.testFloat(1.3f));
 
         assertFalse(filter.testNull());
         assertFalse(filter.testFloat(1.2f));
     }
+
+    @Test
+    public void testMultiRangeWithNaNs()
```

Comment:
This test verifies that expression is converted to proper instance of MultiRange filter, hence, it belongs to TestTupleDomainFilterUtils. Here, you need to add more tests to testMultiRange. These tests should create ranges with nanAllowed = true and confirm that these allow NaNs.

Reply:
I feel nanAllowed=true should not be exposed at all. Because it does not make sense to have a range that does not allow nans but have nanAllowed=true, so this test abstracts it and nanAllowed is set internally based on the range.

Reply:
@bhhari 

- TestTupleDomainFilter is a unit test that verifies correctness of individual TupleDomainFilter implementations
- TestTupleDomainFilterUtils is another unit test that verifier that a TupleDomain is converted to the right TupleDomainFilter instance. This testing is important because different implementations of TupleDomainFilter may have the same behavior, but different performance characteristics. Hence, the need to confirm the specific instance being used.
- The new test is an integration test and as such doesn't belong to either of these tests. It doesn't need to be that. We can (1) add testing for the new nanAllowed flag to TestTupleDomainFilter and (2) add tests that ensure that certain TupleDomains are converted to instance with nanAllowed set to true.